R/ospPlotConfiguration.R

```diff
@@ -0,0 +1,403 @@
+
+    # XAxisConfiguration
+    xAxisLimits = NULL,
+    xAxisScale = NULL,
+    xAxisTicks = NULL,
+    xAxisTickLabels = NULL,
+    xAxisLabelSize = NULL,
+    xAxisLabelColor = NULL,
+    xAxisLabelFontFamily = NULL,
+    xAxisLabelFontFace = NULL,
+    xAxisLabelAngle = NULL,
+
+    # YAxisConfiguration
+    yAxisLimits = NULL,
+    yAxisScale = NULL,
+    yAxisTicks = NULL,
+    yAxisTickLabels = NULL,
+    yAxisLabelSize = NULL,
+    yAxisLabelColor = NULL,
+    yAxisLabelFontFamily = NULL,
+    yAxisLabelFontFace = NULL,
+    yAxisLabelAngle = NULL,
+
+    # background configuration
+    watermark = NULL,
+    watermarkSize = NULL,
+    watermarkColor = NULL,
+    watermarkFontFamily = NULL,
+    watermarkFontFace = NULL,
+    watermarkAngle = NULL,
+    plotBackgroundFill = NULL,
+    plotBackgroundColor = NULL,
+    plotBackgroundSize = NULL,
+    plotBackgroundLinetype = NULL,
+    plotPanelBackgroundFill = NULL,
+    plotPanelBackgroundColor = NULL,
+    plotPanelBackgroundSize = NULL,
+    plotPanelBackgroundLinetype = NULL,
+    xAxisColor = NULL,
+    xAxisSize = NULL,
+    xAxisLinetype = NULL,
+    yAxisColor = NULL,
+    yAxisSize = NULL,
+    yAxisLinetype = NULL,
+    xGridColor = NULL,
+    xGridSize = NULL,
+    xGridLinetype = NULL,
+    yGridColor = NULL,
+    yGridSize = NULL,
+    yGridLinetype = NULL,
+
+    # lines
+    linesColor = NULL,
+    linesFill = NULL,
+    linesShape = NULL,
+    linesSize = NULL,
+    linesLinetype = NULL,
+    linesAlpha = NULL,
+
+    # points
+    pointsColor = NULL,
+    pointsFill = NULL,
+    pointsShape = NULL,
+    pointsSize = NULL,
+    pointsLinetype = NULL,
+    pointsAlpha = NULL,
+
+    # ribbons
+    ribbonsColor = NULL,
+    ribbonsFill = NULL,
+    ribbonsShape = NULL,
+    ribbonsSize = NULL,
+    ribbonsLinetype = NULL,
+    ribbonsAlpha = NULL,
+
+    # errorbars
+    errorbarsColor = NULL,
+    errorbarsFill = NULL,
+    errorbarsShape = NULL,
+    errorbarsSize = NULL,
+    errorbarsLinetype = NULL,
+    errorbarsAlpha = NULL,
+
+    # export
+    plotSaveFileFormat = NULL,
+    plotSaveFileWidth = NULL,
+    plotSaveFileHeight = NULL,
+    plotSaveFileDimensionUnits = NULL,
+
+    #' @description Create a new `ospInternalPlotConfiguration` object
+    #'
+    #' @param title,subtitle,xlabel,ylabel,legendTitle,watermark A character string
+    #'   providing plot annotations for plot title, subtitle, x-axis label, y-axis
+    #'   label, plot legend, watermark, respectively.
+    #' @param legendPosition A character string defining the legend position
+    #'   (default: `"outsideRight"`). Available options
+    #'   are:`"none"`,`"insideTop"`,`"insideTopLeft"`,
+    #'   `"insideLeft"`,`"insideBottomLeft"`,`"insideBottom"`,`"insideBottomRight"`,
+    #'   `"insideRight"`,`"insideTopRight"`,`"outsideTop"`,`"outsideTopLeft"`,
+    #'   `"outsideLeft"`,`"outsideBottomLeft"`,`"outsideBottom"`,`"outsideBottomRight"`,
+    #'    `"outsideRight"`,`"outsideTopRight"`.
```

Comment:
add hint that all options are accessible via `tlf::LegendPositions` e.g. `tlf::LegendPositions$insideLeft`
https://www.open-systems-pharmacology.org/TLF-Library/reference/LegendPositions.html